### PR TITLE
feat: Moonshot supported function call & function call unit tests

### PR DESCRIPTION
**Pull request by mxsl-gr**
PR content:
  1. Moonshot supported function call
  2. function call unit tests

related issue: https://github.com/spring-projects/spring-ai/issues/1058

### CODE REVIEW

**Comment by tzolov** [comment]
HI @mxsl-gr  and thank you for the update. 
As we are deprecating `AbstractFunctionCallSupport` in favor of `AbstractToolCallSupport` would you be interested to re-work your PR to support AbstractToolCallSupport?

For examples, you can check the OpenAI/AzureOpenAI/Anthropic/Mistral/Ollama/VertexAIGemini function calling implementations, or ask me for support ;) 
Would appreciate your help as you own those models.

**Comment by mxsl-gr** [comment]
hi, @tzolov
it worked for now, and this PR has been squash and force pushed

**Comment by markpollack** [comment]
@mxsl-gr thanks, sorry I haven't gotten to merging this yet

**Comment by mxsl-gr** [comment]
> @mxsl-gr thanks, sorry I haven't gotten to merging this yet

hi @markpollack , is ok, i went on a trip and just back now. please let me know if there's anything else to adjust. 

other some PRs related to function calls: https://github.com/spring-projects/spring-ai/pull/1116, https://github.com/spring-projects/spring-ai/pull/1163

File: spring-ai-spring-boot-autoconfigure/src/test/java/org/springframework/ai/autoconfigure/moonshot/tool/FunctionCallbackWrapperIT.java [hunk 1]
```diff
@@ -58,11 +59,12 @@ public class FunctionCallbackWrapperIT {
 
 	@Test
 	void functionCallTest() {
```
Comment (markpollack) [inline spring-ai-spring-boot-autoconfigure/src/test/java/org/springframework/ai/autoconfigure/moonshot/tool/FunctionCallbackWrapperIT.java:61]: This test fails for me.  It looks like it wants to make the calls sequentially or something

```
java.lang.AssertionError: 
Expecting actual:
  "The current weather in San Francisco is 30.0°C. Now, I will check the weather in Tokyo."
to contain:
  ["30", "10", "15"]
but could not find:
  ["10", "15"]
 
```

**Comment by markpollack** [comment]
I'm merging this despite test failure as this PR has improved the support so significantly.  We can circle back to figure out what is going on with the function calling failure in the test.  The streaming function calling test did pass in my testing.

**Comment by markpollack** [comment]
merged in 611c9493e372f96129dd1ada3a0ca82445ba5994

**Pull request closed without merging**